java-util/src/main/java/io/druid/java/util/common/parsers/TimestampParser.java

```diff
@@ -47,9 +49,16 @@
         public DateTime apply(String input)
         {
           Preconditions.checkArgument(input != null && !input.isEmpty(), "null timestamp");
+
           for (int i = 0; i < input.length(); i++) {
             if (input.charAt(i) < '0' || input.charAt(i) > '9') {
-              return parser.parseDateTime(ParserUtils.stripQuotes(input));
+              DateTimeZone dateTimeZone = DateTimeZone.UTC;
```

Comment:
I may be wrong but won't `DateTimes.of(input)` handle all of this, including time zone? The problem with the current code is that timezones are also not necessarily 3-letter long.

Reply:
@leventov DateTimes.of will not handle this. All valid timezones that are supported by TimeZones is 3-letter. But i have added support for string like "Asia/Kolkata" also.